test/Npgsql.Tests/ConnectionTests.cs

```diff
@@ -979,5 +979,53 @@ public void ExceptionDuringClose()
                 conn.Close();
             }
         }
+
+        [Test]
+        [Parallelizable(ParallelScope.None)]
+        public void UsePgPassFile()
+        {
+            var builder = new NpgsqlConnectionStringBuilder(ConnectionString)
+            {
+                Pooling = false,
+                IntegratedSecurity = false,
+                Password = null
+            };
+
+            using (var conn = OpenConnection(builder))
+            {
+                Assert.That(conn.ExecuteScalar("SELECT 1"), Is.EqualTo(1));
+                conn.Close();
+            }
+        }
+
+        #region Setup/Teardown
+
+        string _pgpassEnvVarValue = null;
+        string _pgpassFile = null;
+
+        [OneTimeSetUp]
+        public void CreateTestFile()
```

Comment:
This looks much better than before, but there's still a problem - the setup/teardown runs for all test in the ConnectionTests suite, possibly affecting them. If we have just one test (that's the case), the setup and teardown code should simply be integrated inside the test (the teardown should be in a finally clause). If we have more than one test, it's better to create a different suite with OneTimeSetUp/TearDown as you did, this way everything is properly isolated. Sorry for not making this clear before.